six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp

```diff
@@ -129,14 +129,16 @@ class Tester
         mNormalPathname("normal_write.nitf"),
         mNormalFileCleanup(mNormalPathname),
         mDims(123, 456),
-        mData(createData<DataTypeT>(mDims)),
         mImage(mDims.area()),
         mTestPathname("streaming_write.nitf"),
         mSchemaPaths(schemaPaths),
         mSetMaxProductSize(setMaxProductSize),
         mMaxProductSize(maxProductSize),
         mSuccess(true)
     {
+        // This can't go in the initializer list, because Solaris complains
+        mData = createData<DataTypeT>(mDims);
```

Comment:
Is this a compile time or runtime error on Solaris..?

Reply:
Runtime. It says "Illegal instruction" and aborts.

Reply:
That is really weird (and philosophically annoying).  If you leave it on the initializer list but call `createData.release()`, does that work?

Reply:
Yes!

Reply:
[This](https://docs.oracle.com/cd/E19205-01/820-4155/c++_faq.html#LibComp3) may be related?

Reply:
OK glad we've got a workaround that is at least just slightly off from what we'd normally do.  Hmm, yeah you're probably right that there's some piece of the standard they're not implementing for... whatever reason.